### PR TITLE
Add netrc for cloning with git_credentials

### PR DESCRIPTION
**Pull request by MalloZup**
This is needed when we want to clone from private repo

### CODE REVIEW

**Comment by MalloZup** [comment]
star

**Comment by MalloZup** [comment]
https://github.com/moio/sumaform/pull/198

**Comment by MalloZup** [comment]
@moio ready for a review. 
We need to wait before merging, because it will break the production (since the testsuite dir doesn't exist)
a part of this, i think it is ok

File: modules/libvirt/base/variables.tf [hunk 1]
```diff
@@ -8,6 +8,18 @@ variable "cc_password" {
   type = "string"
 }
 
+variable "git_username" {
```
Comment (moio) [inline modules/libvirt/base/variables.tf]: Can you please move these variables out of `base` and in the `controller` module please? We should not really need them in other hosts.

Comment (MalloZup) [reply modules/libvirt/base/variables.tf]: @moio  ok, i was in doubt were to place them.thx

**Comment by MalloZup** [comment]
@moio 
a part of this, i noticed that having spacewalk in the controller make a lot of positive changes:
- the controller could become the dev machine , where we install `ant` and other tools for deploying src code, adn then run directly the test. 
- we could even imagine to create an ant target where we run some tests

**Comment by MalloZup** [comment]
@moio 
i get right know with the actual commit:

when the controller is runned
```
[ERROR   ] Rendering exception occurred: Jinja variable 'dict object' has no attribute 'hostname'
[CRITICAL] Rendering SLS 'banse:default' failed: Jinja variable 'dict object' has no attribute 'hostname'
local:
    Data failed to compile:
---------    Rendering SLS 'base:default' failed: Jinja variable 'dict object' has no attribute 'hostname'
```
Did you ever had this issue? just for an hint

File: salt/controller/init.sls [hunk 2]
```diff
@@ -99,3 +104,22 @@ extra_pkgs:
       - screen
     - require:
       - sls: controller.repos
+
+git_config:
+  file.append:
+    - name: ~/.netrc
```
Comment (moio) [inline salt/controller/init.sls]: Is the file existing already at this point or are you creating it from scratch?

**Comment by moio** [comment]
> [CRITICAL] Rendering SLS 'banse:default'

ba**n**se looks like a typo

**Comment by MalloZup** [comment]
ahha

File: salt/controller/init.sls (hunk 2 above)
Comment (MalloZup) [reply salt/controller/init.sls]: from scratch

File: modules/libvirt/controller/main.tf [hunk 3]
```diff
@@ -22,6 +22,9 @@ module "controller" {
   ssh_key_path = "${var.ssh_key_path}"
   grains = <<EOF
 
+
+git_username = "${var.git_username}"
```
Comment (MalloZup) [inline modules/libvirt/controller/main.tf]: @moio  here was the typo :rofl: 
``` git_username: ..```

File: salt/controller/init.sls [hunk 4]
```diff
@@ -65,15 +65,20 @@ cucumber_requisites:
     - require:
       - sls: controller.repos
 
-cucumber_testsuite:
+spacewalk_git_repository:
   git.latest:
-    - name: https://github.com/SUSE/spacewalk-testsuite-base
-    - target: /root/spacewalk-testsuite-base
+    # FIXME: this is hardcoded to SUSE(for opensourcing remove
```
Comment (moio) [inline salt/controller/init.sls]: Please remove this FIXME and open an issue instead, so that's easier to track

Comment (MalloZup) [reply salt/controller/init.sls]: ok you are right

**Pull request merged**